api/writer/json.go

```diff
@@ -13,26 +12,32 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-type JSONResponseWriter struct {
+type JSONOrYAMLResponseWriter struct {
+	Type    string
+	Encoder func(io.Writer, interface{}) error
 }
 
-func (j *JSONResponseWriter) start(apiContext *types.APIContext, code int, obj interface{}) {
+func (j *JSONOrYAMLResponseWriter) start(apiContext *types.APIContext, code int, obj interface{}) {
 	AddCommonResponseHeader(apiContext)
-	apiContext.Response.Header().Set("content-type", "application/json")
+	if j.Type == "json" {
```

Comment:
can the type just have j.ContentType and just set `content-type = j.ContentType`